src/main/java/de/tum/cit/aet/artemis/communication/service/ReactionService.java

```diff
@@ -63,12 +64,23 @@ public ReactionService(UserRepository userRepository, CourseRepository courseRep
      * Checks reaction validity, determines the reaction's user,
      * retrieves the associated posting and persists the mutual association
      *
-     * @param courseId if of course the according posting belongs to
-     * @param reaction reaction to create
+     * @param courseId    if of course the according posting belongs to
+     * @param reactionDTO reaction to create
      * @return created reaction that was persisted
      */
-    public Reaction createReaction(Long courseId, Reaction reaction) {
-        Posting posting = reaction.getPost() == null ? reaction.getAnswerPost() : reaction.getPost();
+    public Reaction createReaction(Long courseId, ReactionDTO reactionDTO) {
+        validateReactionAssociation(reactionDTO);
+
+        Reaction reaction = new Reaction();
+        reaction.setEmojiId(reactionDTO.emojiId());
+
+        if (reactionDTO.postId() != null) {
+            reaction.setPost(postRepository.findByIdElseThrow(reactionDTO.postId()));
+        }
+        if (reactionDTO.answerPostId() != null) {
+            reaction.setAnswerPost(answerPostRepository.findByIdElseThrow(reactionDTO.answerPostId()));
+        }
```

Comment:
Okay, so either `postId` or `asnwerPostId` are defined, but never bought? 

Why do we have two fields in the DTO then? Shouldn't it just be one field `relatedPostId` instead? 
Does the client really care if it is a `postId` or an `answerPostId`?